lib/utils/filter.rb

```diff
@@ -6,6 +6,48 @@
 module FilterTable
   module Show; end
 
+  class ExceptionCatcher
+    def initialize(original_resource, original_exception)
+      @original_resource = original_resource
+      @original_exception = original_exception
+    end
+
+    # This method is called via the runner and signals RSpec to output a block
+    # showing why the resource was skipped. This prevents the resource from
+    # being added to the test collection and being evaluated.
+    def resource_skipped?
+      @original_exception.is_a?(Inspec::Exceptions::ResourceSkipped)
+    end
+
+    # This method is called via the runner and signals RSpec to output a block
+    # showing why the resource failed. This prevents the resource from
+    # being added to the test collection and being evaluated.
+    def resource_failed?
+      @original_exception.is_a?(Inspec::Exceptions::ResourceFailed)
+    end
+
+    def resource_exception_message
+      @original_exception.message
+    end
+
+    # Capture message chains and return `ExceptionCatcher` objects
+    def method_missing(*)
+      ExceptionCatcher.new(@original_resource, @original_exception)
```

Comment:
Since we do not keep the chain, could we just return `self`?

Reply:
Yes, we totally can. I was originally misled thinking `self` was going to be the original object that was throwing the exception, but I tricked myself because I was overriding `to_s` and `inspec` in `ExceptionCatcher` to match the original exception for InSpec/RSpec formatter purposes.

I just pushed up a change to change this to `self` instead.

Reply:
Nice :-)